src/govuk/components/select/template.njk

```diff
@@ -42,7 +42,7 @@
   {% for item in params.items %}
     {% if item %}
       <option value="{{ item.value }}"
-        {{-" selected" if item.selected }}
+        {{-" selected" if item.selected or (params.value and item.value == params.value) }}
```

Comment:
I'd recommend checking `item.selected` for undefined, not truthiness. If it's explicitly false you want that stored and it not to fall to the other conditions.